framework/PluginBaseClasses/SupervisedLearningPlugin.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Provides API and utilities for extending the ROM SupervisedLearning interface
+with custom reduced-order model applications.
```

Comment:
Since this targets an API, I would at least list the methods that should be implemented (also because there are additional methods to the one in PluginBase, since the inheritance from SupervisedLearning)

Reply:
I agree with you. This is just a demonstration for the ROM plugin. I think we are lack of documentations for our post-processor, rom, metric and out-stream plugins. I suggest to create an issue for this, and I will address it in another PR. Please let me know what do you think? @aalfonsi